pkg/agent/controller/serviceexternalip/controller.go

```diff
@@ -453,3 +438,19 @@ func (c *ServiceExternalIPController) nodesHasHealthyServiceEndpoint(service *co
 	}
 	return nodes, nil
 }
+
+func (c *ServiceExternalIPController) GetServiceExternalIPStatus() []querier.ServiceExternalIPInfo {
+	c.externalIPStatesMutex.RLock()
+	defer c.externalIPStatesMutex.RUnlock()
+	info := make([]querier.ServiceExternalIPInfo, len(c.externalIPStates))
+	idx := 0
+	for k, v := range c.externalIPStates {
+		info[idx].ServiceName = k.Name
+		info[idx].Namespace = k.Namespace
+		info[idx].ExternalIP = v.ip
+		info[idx].ExternalIPPool = v.ipPool
+		info[idx].AssignedNode = v.assignedNode
+		idx++
+	}
```

Comment:
nit: I think something like this is more elegant:
```go
info := make([]querier.ServiceExternalIPInfo, 0, len(c.externalIPStates))
for k, v := range c.externalIPStates {
    info = append(info, querier.ServiceExternalIPInfo{
        ServiceName: k.Name,
        // ...
    })
```

Reply:
thanks. updated.